### PR TITLE
[GeoMechanicsApplication] Infiltration test with partially saturated Pw-only material

### LINKED ISSUE #14550: [GeoMechanicsApplication] Bringing partially saturated flow tests back to master
A user would like to bring back several validation test to the master. 

**Background**
A list of earlier created tests in KRATOS and DGflow was consulted (see bottom of the issue for reference). From this list, the unsaturated infiltration test (no. 2) is the only relevant Pw-only test regarding partially saturated flow, which therefore will be brought back to master as part of this issue.

Next to that, 4 tests of the staringreeks (B6, B9, O6, O13) are selected as critical test cases to be added as validation test to the Kratos suite.

**Acceptance Criteria**
- [x] Find the tests
- [x] Check with groundwaterflow team which tests need to be brought back (only related to unsaturated flow, saturated optional)
- [ ] Check if tests can be improved/aligned
- [ ] Bring back tests
- [ ] README.md for tests

**For reference**
Image

### PR DESCRIPTION
**Pull request by rfaasse**
## **📝 Description**
This PR adds a Pw-only transient infiltration test and its documentation. Next to this, the plot utilities have been extended slightly to be able to adjust colors manually if needed.

### CODE REVIEW

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/infiltration_from_top_boundary.svg
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/infiltration_from_top_boundary.svg:1]: These results have been validated qualitatively with DG-Flow results. I don't have the data, so it's not in one plot for now, but we can see it's very similar (note the pressure convention is opposite and t1-t4 are the same as the first 4 time steps in our plot):

Image

Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/infiltration_from_top_boundary.svg:1]: Perhaps the water pressure values along the horizontal axis become more readable when we use kPa as unit rather than Pa (since the number of digits will decrease). What do you think?

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/README.md [hunk 1]
```diff
@@ -14,8 +14,25 @@ The simulation is done on a double stage process, and steady state Pw and U-Pw e
...
 
 The picture below illustrates the pressure contours resulting from the simulation (as an example the 2D6N for Pw is shown below).
 
-<img src="documentation_data/test_saturated_below_phreatic_level_pw_triangle6n_results.png" alt="Pressure field for case of saturation below phreatic level at stage 2" title="Pressure field at stage 2" width="600">
\ No newline at end of file
+<img src="documentation_data/test_saturated_below_phreatic_level_pw_triangle6n_results.png" alt="Pressure field for case of saturation below phreatic level at stage 2" title="Pressure field at stage 2" width="600">
+
+## Infiltration test
+
+The geometry and boundary conditions used the infiltration test are shown below. The initial condition is a hydrostatic pressure profile with the reference coordinate at the bottom of the column. This means that at the start, the column contains positive pressures. The boundary condition of $p_w=0$, induces infiltration, which propagates downwards through the column.
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/README.md]: Perhaps the first sentence reads easier when modified as follows:

```suggestion
The geometry and boundary conditions of the infiltration test are shown below. The initial condition is a hydrostatic pressure profile with the reference coordinate at the bottom of the column. This means that at the start, the column contains positive pressures. The boundary condition of $p_w=0$, induces infiltration, which propagates downwards through the column. 
```

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/MaterialParameters.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            "model_part_name": "PorousDomain.sandfine",
+            "properties_id": 1,
+            "Material": {
+                "constitutive_law": {
+                    "name": "GeoLinearElasticPlaneStrain2DLaw"
+                },
+                "Variables": {
+                    "GEO_DRAINAGE_TYPE": "FULLY_COUPLED",
+                    "YOUNG_MODULUS": 1.0e+07,
+                    "POISSON_RATIO": 0.0,
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/MaterialParameters.json:13]: Later, when we have properly separated the Pw-only elements from the U-Pw elements, it would be good to remove these mechanical properties from the material, since they shouldn't be needed, right? No action needed now.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/MaterialParameters.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                    "DENSITY_SOLID": 2.0e+03,
+                    "DENSITY_WATER": 1.0e+03,
+                    "POROSITY": 0.33,
+                    "BULK_MODULUS_SOLID": 6.7e+19,
+                    "BULK_MODULUS_FLUID": 1.0e+20,
+                    "PERMEABILITY_XX": 1.157e-13,
+                    "PERMEABILITY_YY": 1.157e-13,
+                    "PERMEABILITY_XY": 0.0,
+                    "DYNAMIC_VISCOSITY": 1.0e-03,
+                    "THICKNESS": 1.0,
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/MaterialParameters.json]: I would expect to not need to specify a `"THICKNESS"`, since we're adopting a plane strain model, which has an implicit thickness of 1.0 length unit. Can we remove this input parameter here?

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        "model_import_settings": {
+            "input_type": "mdpa",
+            "input_filename": "run1sim5"
+        },
+        "material_import_settings": {
+            "materials_filename": "MaterialParameters.json"
+        },
+        "time_stepping": {
+            "time_step": 600.0,
+            "max_delta_time_factor": 1
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json]: I think this should be a floating point value rather than an integer value:

```suggestion
            "max_delta_time_factor": 1.0
```

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        "scheme_type": "Backward_Euler",
+        "strategy_type": "newton_raphson",
+        "convergence_criterion": "residual_criterion",
+        "residual_relative_tolerance": 1.0e-04,
+        "residual_absolute_tolerance": 1.0e-09,
+        "min_iterations": 2,
+        "max_iterations": 15,
+        "number_cycles": 2,
+        "reduction_factor": 1.0,
+        "increase_factor": 1.0,
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json]: I may be wrong, but why is `"number_cycles"` greater than 1 here? If I remember correctly, cycles are relevant in the context of adaptive time stepping. But both "`reduction_factor"` and `"increase_factor"` are set to 1.0, which means that we will attempt one and the same time step increment for all cycles. If the step doesn't converge in the first cycle, there is no point in attempting a second cycle with the same time step increment.

```suggestion
        "number_cycles": 1,
        "reduction_factor": 1.0,
        "increase_factor": 1.0,
```

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+        "desired_iterations": 2,
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json]: If I remember rightly, this is an input parameter for arc-length control. That's not needed here.

```suggestion
```

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+                                "GiDPostMode": "GiD_PostAscii",
+                                "MultiFileFlag": "SingleFile"
+                            },
+                            "file_label": "step",
+                            "output_control_type": "step",
+                            "output_interval": 1,
+                            "body_output": true,
+                            "node_output": false,
+                            "skin_output": false,
+                            "plane_output": [],
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json]: I think we can safely remove this input parameter, since we don't need it.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+                            "nodal_results": ["WATER_PRESSURE"],
+                            "gauss_point_results": []
+                        },
+                        "point_data_configuration": []
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json]: Also this input parameter can be removed safely, I think.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+                "python_module": "apply_scalar_constraint_table_process",
+                "kratos_module": "KratosMultiphysics.GeoMechanicsApplication",
+                "process_name": "ApplyScalarConstraintTableProcess",
+                "Parameters": {
+                    "model_part_name": "PorousDomain.topP",
+                    "variable_name": "WATER_PRESSURE",
+                    "is_fixed": true,
+                    "fluid_pressure_type": "Uniform",
+                    "value": 0.0,
+                    "table": 1
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json:112]: I have cross-checked the table with ID 1, and even though it's technically correct (it will interpolate from 0.0 to 0.0), I think this is rather cumbersome way of prescribing a water pressure that's equal to 0.0 on all the nodes of the model part. I would suggest to simplify the condition, and just use a value of 0.0, and to not use a table here.

```suggestion
                    "value": 0.0,
                    "table": 0
```

(Obviously, then also the table needs to be removed from the MDPA file.)

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+                "Parameters": {
+                    "model_part_name": "PorousDomain.porous_computational_model_part",
+                    "variable_name": "VOLUME_ACCELERATION",
+                    "active": [false, true, false],
+                    "value": [0.0, -10.0, 0.0],
+                    "table": [0, 0, 0]
+                }
+            }
+        ],
+        "auxiliar_process_list": []
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json]: We can remove this empty process list.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+                "kratos_module": "KratosMultiphysics.GeoMechanicsApplication",
+                "process_name": "ApplyVectorConstraintTableProcess",
+                "Parameters": {
+                    "model_part_name": "PorousDomain.porous_computational_model_part",
+                    "variable_name": "VOLUME_ACCELERATION",
+                    "active": [false, true, false],
+                    "value": [0.0, -10.0, 0.0],
+                    "table": [0, 0, 0]
+                }
+            }
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json:139]: Just out of curiosity, do we need really need this process (in terms of the kind of simulation that we're doing)? Or is it somehow required, since the Pw-only elements are still strongly coupled to the U-Pw elements (through inheritance)?

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/run1sim5.mdpa [hunk 12]
```diff
@@ -0,0 +1,1305 @@
+Begin Table 1 TIME WATER_PRESSURE
+   0.000000e+00   -0.000000e+00
+   4.800000e+04   -0.000000e+00
+End Table
+
+Begin Table 2 TIME NORMAL_FLUID_FLUX
+   0.000000e+00    0.000000e+00
+   4.800000e+04    0.000000e+00
+End Table
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/run1sim5.mdpa]: As mentioned before, I don't think we need the table with ID 1. The table with ID 2 is not referenced by any process, so that one is for sure redundant for the test case at hand.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/run1sim5.mdpa [hunk 13]
```diff
@@ -0,0 +1,1305 @@
...
+
+
+Begin Conditions PwNormalFluxCondition2D2N
+  1 1  3  4 
+End Conditions
+
+
+Begin SubModelPart sandfine
+  Begin SubModelPartTables
+  End SubModelPartTables
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/run1sim5.mdpa]: We can remove this empty list of tables.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/run1sim5.mdpa [hunk 14]
```diff
@@ -0,0 +1,1305 @@
...
+    340
+    341
+    342
+    343
+    344
+    345
+    346
+  End SubModelPartElements
+  Begin SubModelPartConditions
+  End SubModelPartConditions
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/run1sim5.mdpa]: Also this empty list of conditions can be removed.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/run1sim5.mdpa [hunk 15]
```diff
@@ -0,0 +1,1305 @@
...
+ End SubModelPart
+
+
+Begin SubModelPart topP
+  Begin SubModelPartTables
+    1
+  End SubModelPartTables
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/run1sim5.mdpa:1276]: If we use a fixed value of 0.0 rather than a table for the water pressure along the top of the domain, this list of tables can be removed.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/run1sim5.mdpa [hunk 16]
```diff
@@ -0,0 +1,1305 @@
...
+    1
+    2
+  End SubModelPartNodes
+End SubModelPart
+
+
+Begin SubModelPart botF
+  Begin SubModelPartTables
+    2
+  End SubModelPartTables
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/run1sim5.mdpa]: The table with ID 2 is listed here, but nowhere else. Since the table is effectively not being used, I would suggest to remove this list, too.

File: applications/GeoMechanicsApplication/tests/test_partial_saturation.py
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partial_saturation.py:1]: Just a general remark: I've noticed that some of the changes in this file are due to formatting it with `black`. However, for Python files, we don't have formatting integrated in `kp format`. Consequently, when this file is edited by someone who doesn't run `black` afterwards, the formatting will become inconsistent again. Should we discuss with our PO to prioritize an extension of `kp format` for Python files, just so we can gradually introduce a consistent style for Python scripts (with an exception list and everything, just like we have for JSON files)? Curious what you think.

File: applications/GeoMechanicsApplication/tests/test_partial_saturation.py [hunk 17]
```diff
@@ -45,52 +55,218 @@ def __test_saturated_below_phreatic_level_pw(self, test_name):
...
+        class ExpectedResult:
+            node_id: int
+            value: float
+
+        expected_results_at_times = {
+            12000: [
+                ExpectedResult(node_id=1, value=0.0),
+                ExpectedResult(node_id=17, value=6243.59),
+                ExpectedResult(node_id=26, value=16400),
+            ],
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partial_saturation.py]: This is very nice and easy to read. 👍 Thanks!

File: applications/GeoMechanicsApplication/tests/test_partial_saturation.py [hunk 18]
```diff
@@ -45,52 +55,218 @@ def __test_saturated_below_phreatic_level_pw(self, test_name):
...
+        ).Nodes:
+            if node.X == 0.0:
+                depth_by_id_for_left_boundary_nodes[node.Id] = -1.0 * node.Y
+
+        @dataclass
+        class ExpectedResult:
+            node_id: int
+            value: float
+
+        expected_results_at_times = {
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partial_saturation.py:169]: Since times are floating point values, I would prefer to use that rather than integer values. For instance, the key of the first item then becomes 12000.0 rather than 12000.

File: applications/GeoMechanicsApplication/tests/test_partial_saturation.py [hunk 19]
```diff
@@ -45,52 +55,218 @@ def __test_saturated_below_phreatic_level_pw(self, test_name):
...
+            12000: [
+                ExpectedResult(node_id=1, value=0.0),
+                ExpectedResult(node_id=17, value=6243.59),
+                ExpectedResult(node_id=26, value=16400),
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partial_saturation.py]: Nitpicking:

```suggestion
                ExpectedResult(node_id=26, value=16400.0),
```

File: applications/GeoMechanicsApplication/tests/test_partial_saturation.py [hunk 20]
```diff
@@ -45,52 +55,218 @@ def __test_saturated_below_phreatic_level_pw(self, test_name):
...
+                ExpectedResult(node_id=87, value=-293.249),
+                ExpectedResult(node_id=91, value=970.378),
+                ExpectedResult(node_id=100, value=1672.53),
+            ],
+            192000: [
+                # This is the hydrostatic line from p = 0 at depth = 0m
+                # and p = -20000 Pa at depth = 2m, as seen in the plot
+                ExpectedResult(node_id=1, value=0.0),
+                ExpectedResult(node_id=58, value=-10000),
+                ExpectedResult(node_id=3, value=-20000),
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partial_saturation.py]: Sorry, more nitpicking:

```suggestion
                ExpectedResult(node_id=58, value=-10000.0),
                ExpectedResult(node_id=3, value=-20000.0),
```

File: applications/GeoMechanicsApplication/tests/test_partial_saturation.py [hunk 21]
```diff
@@ -45,52 +55,218 @@ def __test_saturated_below_phreatic_level_pw(self, test_name):
...
+                water_pressures, expected_water_pressures, places=None, delta=10.0
+            )
+
+        if test_helper.want_test_plots():
+            self.create_pressure_depth_plots(
+                depth_by_id_for_left_boundary_nodes,
+                expected_results_at_times,
+                file_path,
+                output_data,
+            )
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/test_partial_saturation.py]: You may want to do the plotting before asserting the test results. If the latter fails, now no plots will be generated, which can actually be quite handy when investigating a test failure.

**Review by avdg81 — COMMENTED** [review]
Hi @rfaasse and @mnabideltares,
Thanks a lot for creating this infiltration test with a partially saturated material. I'm glad to see that the test can generate plots of the water pressure distribution over time. That really helps to understand the physical problem that we're trying to solve.
My suggestions are all (very) minor and non-blocking.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/README.md (hunk 1 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/README.md]: Good catch, done!

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json (hunk 5 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json]: Good point, changed!

File: applications/GeoMechanicsApplication/tests/test_partial_saturation.py
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_partial_saturation.py:1]: Yes, good idea to have at least new files formatted with black (same with the approach for json) 👍

File: applications/GeoMechanicsApplication/tests/test_partial_saturation.py (hunk 18 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_partial_saturation.py:169]: Agreed, done!

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json (hunk 11 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json:139]: Pw-only calculations still take gravity into account. It's part of the Pw right hand side (see e.g. the [FluidBodyFlowCalculator ](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/GeoMechanicsApplication/custom_elements/contribution_calculators/fluid_body_flow_calculator.hpp#L70)for our new PwElement)

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json [hunk 22]
```diff
@@ -0,0 +1,147 @@
...
+                            },
+                            "file_label": "step",
+                            "output_control_type": "step",
+                            "output_interval": 1,
+                            "body_output": true,
+                            "node_output": false,
+                            "skin_output": false,
+                            "plane_output": [],
+                            "nodal_results": ["WATER_PRESSURE"],
+                            "gauss_point_results": []
```
Comment (markelov208) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json]: I think this line also can be removed. As far as I remember Kratos has this as default.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/infiltration_from_top_boundary.svg
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/infiltration_from_top_boundary.svg:1]: Nice idea, done!

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/MaterialParameters.json (hunk 3 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/MaterialParameters.json]: Removed!

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json (hunk 4 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json]: Done!

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json (hunk 6 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json]: Indeed, missed this one in the clean-up

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json (hunk 9 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json:112]: As discussed the 'constant' and time-dependent versions of the processes don't behave the same. I'll create an issue to investigate, and we'll keep it like this for now.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json (hunk 10 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json]: Done!

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/run1sim5.mdpa (hunk 12 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/run1sim5.mdpa]: As discussed, will keep table 1, removed table 2

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/run1sim5.mdpa (hunk 15 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/run1sim5.mdpa:1276]: As discussed,keeping it like this. To be investigated in a separate issue

File: applications/GeoMechanicsApplication/tests/test_partial_saturation.py (hunk 21 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_partial_saturation.py]: Agreed, moved!

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json (hunk 22 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json]: Indeed, removed!

**Review by rfaasse — COMMENTED** [review]
Thanks for the suggestions! I processed them, so the PR is ready for re-review

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/infiltration_from_top_boundary.svg
Comment (avdg81) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/infiltration_from_top_boundary.svg:1]: Yes, this looks really nice. Thanks for changing the unit. 👍

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json (hunk 9 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json:112]: One minor request: can you please add the link to the new issue (after you've created it) as a comment here, just so we can easily find it? Thanks. 👍

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json (hunk 11 above)
Comment (avdg81) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_infiltration_pw/ProjectParameters.json:139]: Ah, I see. Thanks for pointing this out. 👍

**Review by avdg81 — APPROVED** [review]
Hi Richard,
Thanks for taking care of the review suggestions. I think this PR is ready to go. 👍

**Review by mnabideltares — APPROVED** [review]
OK to me. Thank you
